tools/pipesys/src/cmd/link.rs

```diff
@@ -0,0 +1,194 @@
+use super::fetch_fds;
+
+use anyhow::{bail, Context, Result};
+use clap::Parser;
+use daemonize::{Daemonize, Outcome};
+use futures::{Future, StreamExt};
+use inotify::{Inotify, WatchMask};
+use log::{error, info, trace};
+use std::path::{Path, PathBuf};
+use std::{env, process};
+use tokio::fs;
+
+#[derive(Debug, Parser)]
+pub(crate) struct Link {
+    /// Fetch the file descriptor for a path from this abstract socket.
+    #[clap(long = "fd-socket")]
+    fd_socket: String,
+
+    /// Create this target path as a link to the file descriptor.
+    #[clap(long = "target")]
+    target: PathBuf,
+}
+
+impl Link {
+    /// Retrieve the file descriptor, then spawn a background process to create the link so that it
+    /// survives the return of the foreground process.
+    pub(crate) async fn execute(&self) -> Result<()> {
+        // Remove the existing symlink, if present.
+        let _ = fs::remove_file(&self.target).await;
```

Comment:
Should we be more defensive when we remove the target?

I'm not sure if we can assert that it is a pipesys-managed link here, but we could maybe refuse to delete non-symlinks.